torch/export/_unlift.py

```diff
@@ -14,6 +15,29 @@
 )
 
 
+@torch._dynamo.disable
+def _check_input_constraints_pre_hook(self, *args, **kwargs):
+    if self._in_spec is not None:
+        try:
+            args = fx_pytree.tree_flatten_spec(
+                args, self._in_spec, exact_structural_match=True
+            )  # type: ignore[assignment]
```

Comment:
@suo I had to change this back to tree_flatten_spec because it's able to handle cases where we pass in kwargs with different orders than how it was exported (ex. {"kwarg1": 1, "kwarg2":2} vs. {"kwarg2": 2, "kwarg1": 1}) . Do you have a better suggestion?

Reply:
strictly speaking, what you described should be an error:
<img width="275" alt="image" src="https://github.com/pytorch/pytorch/assets/1617424/5dbf1121-2e6b-45c9-8fdb-73d64a8a832d">

Reply:
either way, not supporting custom pytrees is a bigger downside than not being relaxed about dict ordering, so still not sold on tree_flatten_spec here.